runtime/drivers/olap.go

```diff
@@ -450,6 +450,20 @@ func (d Dialect) DateTruncExpr(dim *runtimev1.MetricsViewSpec_DimensionV2, grain
 	}
 }
 
+// unit - ie second, minute, ...
+func (d Dialect) DateDiff(unit string, t1, t2 time.Time) (string, error) {
```

Comment:
Might make sense to accept `runtimev1.TimeGrain` here instead and call `d.ConvertToDateTruncSpecifier` internally